### ISSUE TITLE
[macOS] - Compile error when using Folly 

### ISSUE THREAD

**Issue by Allan-Gong** [facebook/folly#1673]
Folly is successfully installed on mac via brew:
```
$ brew install folly
```

However, encountered the following compiling error when trying to use folly:
```
#include <folly/String.h>

......

std::string str = "test1,test2";
std::vector<std::string> tokens;
folly::split(",", str, tokens);
```

Compile error:
```
> Executing task: /usr/bin/clang++ -std=c++17 -stdlib=libc++ -g /Users/gonghany/dev/c++_playground/*.cpp -o /Users/gonghany/dev/c++_playground/test <

In file included from /Users/gonghany/dev/c++_playground/test.cpp:4:
In file included from /usr/local/include/folly/String.h:27:
In file included from /usr/local/include/folly/Conv.h:116:
In file included from /usr/local/include/folly/Demangle.h:19:
In file included from /usr/local/include/folly/FBString.h:33:
In file included from /usr/local/include/folly/CPortability.h:22:
In file included from /usr/local/include/folly/portability/Config.h:20:
/usr/local/include/folly/folly-config.h:22:34: warning: 'TARGET_OS_SIMULATOR' is not defined, evaluates to 0 [-Wundef-prefix=TARGET_OS_]
     (TARGET_IPHONE_SIMULATOR || TARGET_OS_SIMULATOR || TARGET_OS_IPHONE))
                                 ^
/usr/local/include/folly/folly-config.h:22:57: warning: 'TARGET_OS_IPHONE' is not defined, evaluates to 0 [-Wundef-prefix=TARGET_OS_]
     (TARGET_IPHONE_SIMULATOR || TARGET_OS_SIMULATOR || TARGET_OS_IPHONE))
                                                        ^
2 warnings generated.
Undefined symbols for architecture x86_64:
  "folly::makeConversionError(folly::ConversionCode, folly::Range<char const*>)", referenced from:
      std::__1::enable_if<!(std::is_same<folly::Range<char const*>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >::value), std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >::type folly::to<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(folly::Range<char const*>)::'lambda'(folly::ConversionCode)::operator()(folly::ConversionCode) const in test-96191f.o
      std::__1::enable_if<!(std::is_same<folly::Range<char const*>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >::value), std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >::type folly::to<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >(folly::Range<char const*>)::'lambda0'(folly::ConversionCode)::operator()(folly::ConversionCode) const in test-96191f.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
The terminal process "zsh '-c', '/usr/bin/clang++ -std=c++17 -stdlib=libc++ -g /Users/gonghany/dev/c++_playground/*.cpp -o /Users/gonghany/dev/c++_playground/test'" terminated with exit code: 1.

Terminal will be reused by tasks, press any key to close it.
```

Please help

**Comment by Orvid** [comment]
You need to actually link folly in in order to use it.

**Comment by wAuner** [comment]
You need to provide all the necessary linker flags and header search paths. For me (installed via homebrew on an M1 mac, Intel has different paths) it was:
```
# header search paths 
-I /opt/homebrew/Cellar/folly/2022.03.21.00_1/include 
-I /opt/homebrew/Cellar/boost/1.78.0_1/include 
-I /opt/homebrew/Cellar/glog/0.6.0/include 
-I /opt/homebrew/Cellar/gflags/2.2.2/include 
-I /opt/homebrew/Cellar/fmt/8.1.1_1/include 
# linker search paths
-L/opt/homebrew/Cellar/folly/2022.03.21.00_1/lib/static 
-L/opt/homebrew/Cellar/fmt/8.1.1_1/lib 
# link libraries
-lfolly -lfmt
```

**Comment by barracuda156** [comment]
We are getting this error on 10.6<OS<10.15 in Macports now:
```
folly/work/build/folly/folly-config.h:22:34: error: 'TARGET_OS_SIMULATOR' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]
     (TARGET_IPHONE_SIMULATOR || TARGET_OS_SIMULATOR || TARGET_OS_IPHONE))
                                 ^
folly/work/build/folly/folly-config.h:22:57: error: 'TARGET_OS_IPHONE' is not defined, evaluates to 0 [-Werror,-Wundef-prefix=TARGET_OS_]
     (TARGET_IPHONE_SIMULATOR || TARGET_OS_SIMULATOR || TARGET_OS_IPHONE))
                                                        ^
2 errors generated.
```

**Comment by Orvid** [comment]
This should have been fixed by https://github.com/facebook/folly/commit/2fe2b638066face08aaa6fad6c896d89397e40b3 last week. Are you still seeing this even with that change?

**Comment by barracuda156** [comment]
> This should have been fixed by [2fe2b63](https://github.com/facebook/folly/commit/2fe2b638066face08aaa6fad6c896d89397e40b3) last week. Are you still seeing this even with that change?

Thank you! I referenced your fix in our Macports thread.

For now, this worked for v. 2022.08.08: https://github.com/macports/macports-ports/commit/9aefc182695dbb172fe34602361f53c36f6f2f77
